cmd/soroban-cli/src/commands/contract/arg_parsing.rs

```diff
@@ -63,12 +68,20 @@ pub fn build_host_function_parameters(
     cmd.build();
     let long_help = cmd.render_long_help();
 
-    // get_matches_from exits the process if `help`, `--help` or `-h`are passed in the slop
-    // see clap documentation for more info: https://github.com/clap-rs/clap/blob/v4.1.8/src/builder/command.rs#L631
-    let mut matches_ = cmd.get_matches_from(slop);
-    let Some((function, matches_)) = &matches_.remove_subcommand() else {
-        println!("{long_help}");
-        std::process::exit(1);
+    // try_get_matches_from returns an error if `help`, `--help` or `-h`are passed in the slop
+    // see clap documentation for more info: https://github.com/clap-rs/clap/blob/v4.1.8/src/builder/command.rs#L586
+    let maybe_matches = cmd.try_get_matches_from(slop);
+    let Some((function, matches_)) = (match maybe_matches {
+        Ok(mut matches) => &matches.remove_subcommand(),
+        Err(e) => {
+            // to not exit immediately (to be able to fetch help message in tests), check for an error
+            if e.kind() == DisplayHelp {
+                return Ok(HelpMessage(e.to_string()));
+            }
+            e.exit();
+        }
+    }) else {
+        return Ok(HelpMessage(format!("{long_help}")));
```

Comment:
Instead of passing the Help message up through the stack as an "Ok", can we wrap it in an error, and then unpack that error at the top level before exit is called? That's the pattern clap uses, see https://docs.rs/clap/latest/clap/error/enum.ErrorKind.html#variant.DisplayHelp in its own errors, and is much less confusing than adding a third channel of communication because we'd still have `(Ok, Err)` instead of `((Left, Right), Err)`.

Reply:
My thoughts were to indicate that it can either return help message or some result (both should be handled to not return error code) -- but I agree that following error pattern is better readability